File: LeetCode/3379/golang/lc3379.go
package lc3379

// Runtime complexity: O(n)
// Auxiliary space: O(n) (for the answer array. Except that, constant space is used.)
// Subjective level: easy
// Solved on: 2026-02-05
func constructTransformedArray(nums []int) []int {
	n := len(nums)
	ans := make([]int, n)
	for i := range nums {
		sourceIndex := (i + nums[i] + (1000 * n)) % n
		ans[i] = nums[sourceIndex]
	}
	return ans
}

File: LeetCode/3379/python/lc3379.py
from typing import List

# Runtime complexity: O(n)
# Auxiliary space complexity: O(n)
# Subjective level: easy
# Solved on: 2026-02-05
class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = [0] * n
        for i in range(n):
            ans[i] = nums[(i + nums[i] + (1000*n)) % n]

        return ans
